File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/anagrafiche/Campo.cs
using System;
using System.Collections.Generic;
using System.Text;
using AgronicaCoreModelsSTD.analisi;
using AgronicaCoreModelsSTD.metaschema.utilizzi;

namespace AgronicaCoreModelsSTD.anagrafiche
{
    public class Campo 
    {

        public string descrizione { get; set; }
        public IntervalloTemporale validita { get; set; }
        public string campo_Codice { get; set; }
        public int orientamento_Colturale { get; set; }

        public Specie specie { get; set; }

        public bool serra { get; set; }

        public List<CodiciAnagrafeValori> codici { get; set; }

        public List<CatastoCampo> catastoCampo { get; set; }

        public List<AppezzamentoCampo> appezzamentoCampo { get; set; }

        /// <summary>
        /// dato cartografico associato
        /// </summary>
        public string cartografia { get; set; }

        /// <summary>
        /// flag identificativo se l'appezzamento è disegnato tramite tramite il gps
        /// </summary>
        public bool flag_gps { get; set; }

        //public List<Codici> Codici { get; set; }
        //public List<Catasto_Campo> Catasto { get; set; }

        public PK primaryKey { get; set; }

        public bool flag_cancellazione { get; set; }

        public Campo (PK primaryKey)
        {
            this.primaryKey = primaryKey;
            this.flag_cancellazione = false;
        }

        public Campo()
        {
            flag_cancellazione = false;
        }

        public class PK
        {
            public CentroAziendale.PK centroAziendalePK { get; set; }

            public int codice { get; set; }

            public PK(int codice,  CentroAziendale.PK centroAziendalePK)
            {
                this.codice = codice;
                this.centroAziendalePK = centroAziendalePK;
            }

            public PK()
            {

            }

        }
    }


}

File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/anagrafiche/Campo.ts
import { AppezzamentoCampo } from 'app/anagrafica/campi/campi-edit/appezzamento-campo-edit/appezzamento-campi-edit';
import { Specie } from '../metaschema/utilizzi/Specie';
import { CatastoCampo } from './CatastoCampo';
import { PKCentroAziendale } from './CentroAziendale';
import { CodiciAnagrafeValori } from './CodiciAnagrafeValori';
import { IntervalloTemporale } from './IntervalloTemporale';

export type PKCampo = typeof Campo.PK.prototype;
export class Campo {

    descrizione: string;
    validita: IntervalloTemporale;
    campo_Codice: string;
    orientamento_Colturale: number;
    catastoCampo: CatastoCampo[];
    primaryKey: PKCampo;
    codici: CodiciAnagrafeValori[];
    specie: Specie;
    serra: boolean;
    cartografia: string;
    flag_gps: boolean;

    appezzamentoCampo: AppezzamentoCampo[];

    flag_cancellazione: boolean;

    constructor(primaryKey: PKCampo) {
        this.primaryKey = primaryKey;
        this.flag_cancellazione = false;
    }

    public static PK = class {
        centroAziendalePK: PKCentroAziendale;
        codice: number;

        constructor(codice: number, centroAziendalePK: PKCentroAziendale) {
            this.codice = codice;
            this.centroAziendalePK = centroAziendalePK;
        }
    };

    static Empty(piva: string, sa_cod: number = 0) {
        const campo = new Campo(new this.PK(0, { codice: sa_cod, partitaIva: piva}));
        campo.catastoCampo = [];
        campo.codici = [];
        campo.specie = new Specie(0);
        campo.validita = new IntervalloTemporale();
        return campo;
    }

}


